actors/miner/src/types.rs

```diff
@@ -145,6 +148,53 @@ pub struct ExpirationExtension {
     pub sectors: UnvalidatedBitField,
     pub new_expiration: ChainEpoch,
 }
+#[allow(clippy::too_many_arguments)] // validate mut prevents implementing From
+impl Into<ValidatedExpirationExtension> for ExpirationExtension2 {
+    fn into(mut self) -> ValidatedExpirationExtension {
+        let mut sectors = BitField::new();
+        for sc in self.sectors_with_claims {
+            sectors.set(sc.sector_number)
+        }
+        sectors |= self.sectors.validate_mut().unwrap();
+
+        ValidatedExpirationExtension {
+            deadline: self.deadline,
+            partition: self.partition,
+            sectors,
+            new_expiration: self.new_expiration,
+        }
+    }
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
```

Comment:
For most other params/return objects we derive much more than this. Please copy derives from, e.g. verifreg. I don't think code size matters as unused methods will be dropped for the WASM build. But implementing display, debug, clone etc is often useful for development or debugging.